dnf/automatic/main.py

```diff
@@ -367,9 +368,13 @@ def main(args):
                 exit_code = os.waitstatus_to_exitcode(os.system(conf.commands.reboot_command))
                 if exit_code != 0:
                     logger.error('Error: reboot command returned nonzero exit code: %d', exit_code)
+                    emitters.notify_error('Error: reboot command returned nonzero exit code: %d', exit_code)
+                    emitters.commit()
                     return 1
     except dnf.exceptions.Error as exc:
         logger.error(_('Error: %s'), ucd(exc))
+        emitters.notify_error(_('Error: %s') % str(exc))
+        emitters.commit()
```

Comment:
`emitters` are created quite late in the `try/except` block. You should either test that `emitters is not None` here, or call `build_emitters()` earlier.

Reply:
After looking it over, the build_emitters function throws a ConfigError if none of the options match which would cause the catch statement to invoke. In this case the emitters would be None. I added a check to see if emitters is not None before calling the commit function.